### ISSUE TITLE
The U-MLP model

### ISSUE THREAD

**Issue by ZhouCX117** [twoertwein/UncertaintyRegression#1]
Thank you for your excellent work. I'm a bit confused about which test code is corresponding to the U-MLP model.
And I don't know how to reuse your code.

**Comment by twoertwein** [comment]
Sorry, back then I didn't release the training functions+dataloader. I only shared the building blocks from my shared code repository. 

I will be working on having a full working example for all the methods here soon (probably on just MNIST for simplicity).

U-MLP is just a normal MLP that tries to predict the loss of a primary model. There is nothing special about its architecture. In my experience (and what was done in the paper) it works best if the input to the secondary model (the U-MLP) is a representation from the primary model concatenated with the prediction of the primary model.

> And I don't know how to reuse your code.

I intended that people can instantiate the linked classes, e.g. for the [sparse Gaussian Process](https://bitbucket.org/twoertwein/python-tools/src/e98707bbb102f775b1c9632bf8a0c72af83c9af1/ml/neural.py#lines-736). And then simply integrate that in their pipeline.

**Comment by twoertwein** [comment]
I found my old code. I will have to do some cleaning and slight restructuring. After making sure that it returns the reported results, I will then push the code to this repository. I expect to push it this weekend.

**Comment by twoertwein** [comment]
@ToBeNormal I pushed the training/grid-search functions yesterday.

My next ToDos are (probably by the end of the next weekend): 1) functions to get the metrics reported in the paper; 2) share the data for the dataloaders (get the mnist data from `torchvision` instead of from a pickle file; upload data from DISFA/BP4D+); and 3) update the README with examples how to reproduce the results from the paper.

**Comment by DominickZhang** [comment]
Hi,

Many thanks for providing the codes. I am a bit confused about the command for training the secondary model in README. Isn't the executed file named "train_secundary.py"?

**Comment by twoertwein** [comment]
Thank you, I updated the readme.
